src/resolvers/Mutation/createPost.ts

```diff
@@ -93,6 +94,38 @@ export const createPost: MutationResolvers["createPost"] = async (
     imageUrl: args.file ? uploadImageObj?.newImagePath : "",
   });
 
+  // Check if the pinned property is set
+  if (args.data.pinned) {
```

Comment:
Make this `args.data.pinned === true`. We don't need to check all other cases.

Reply:
The current implementation and one you are suggesting are equivalent. With your approach, we would only get the true value if `true === true`, which is a redundant check, JS will treat `null`, `undefined` and `false` all to be `falsy` by deafult.